bank/README.md

```diff
@@ -0,0 +1,411 @@
+Accurate, reliable interbank transfers [predate widespread electronic systems by over a century](https://cacm.acm.org/opinion/victorian-data-processing/).
+Many features (such as the ability to write checks on paper) along with associated artifacts (bad checks, disputed charges, and processes to deal with them) remain in modern banking.
+So, banking remains a system of requests, settlements/reconciliation, and compensating actions, rather than a system of near-instant ACID transactions.
+
+### DBOS Bank Transfer Model
+
+Many interbank transactions rely on a third-party [clearing house](https://en.wikipedia.org/wiki/Automated_clearing_house).
+However, this DBOS Bank Demo App acts more like a [wire transfer](https://en.wikipedia.org/wiki/Wire_transfer), where banks send messages to each other directly.
+
+Of course the demo is simplified: it does not use [SWIFT](https://en.wikipedia.org/wiki/SWIFT) or a similar standard network for messaging, nor is the settlement process (wherein the sending bank transfers some real assets to the receiving bank) implemented.
+
+The demo does, however:
+
+* Verify that the sending account has funds available to send
+* Put a hold on funds during the course of transfer
+* Send the transfer request to another bank server
+* Credit the recipients account and deduct from the sender's account
+
+The DBOS Bank Demo App ensures that either both accounts are affected, or neither is, regardless of any functional or non-functional errors.  This is done with very little error-handling code, and the error-handling code that exists is in support of business requirements (accounts must exist, sender must actually have the money, etc.) rather than for handling hardware or environmental issues.  This means that all the error handling code in DBOS Bank can be easily tested in automated tests; environmental errors are handled by the framework.
+
+## Run the Demo Locally
+
+### Start PostgreSQL
+
+First, let's set up a database for each bank.  Each bank app can have its own database server, but, to simplify deployment, we will put both bank databases under the same Postgres server process.
+
+The demo provides two ways to set up Postgres:
+* A script set up a Docker container with Postgres configured
+* Use of a preexisting Postgres server
+
+#### Using Docker for Postgres
+
+Set the `PGPASSWORD` environment variable to whatever you'd like, then start Postgres in a Docker container using the `start_postgres_docker` script:
+```shell
+export PGPORT=5432 # Optional; can be set to a non-default value, but make adjustments below
+export PGPASSWORD=<database password>
+./scripts/start_postgres_docker.sh
+```
+This script sets up two new Postgres users: `bank_a` that owns the database `bank_a`, and `bank_b` that owns database `bank_b`.
```

Comment:
Mention that our script uses the same `$PGPASSWORD` for both banks. But the passwords can be changed in the script.